sanic/response.py

```diff
@@ -249,6 +192,15 @@ def text(
     :param headers: Custom Headers.
     :param content_type: the content type (string) of the response
     """
+    # Type conversions are deprecated and quite b0rked but still supported for
+    # text() until applications get fixed. This try-except should be removed.
+    try:
+        # Avoid repr(body).encode() b0rkage for body that is already encoded.
+        # memoryview used only to test bytes-ishness.
+        with memoryview(body):
+            pass
+    except TypeError:
+        body = f"{body}"  # no-op if body is already str
```

Comment:
We should think about throwing a `warning` here.

Reply:
@Tronic can you also take a look at the conflict in `headers.py`?

Reply:
Which types should be deprecated for `text()`?

- [ ] str
- [ ] bytes
- [ ] numeric and others with meaningful `__str__`
- [ ] those that produce `<Foo object at 0x106471210>`

Reply:
Deprecation warning added for everything but str-derived types.